src/tests/tables.test.js

```diff
@@ -20,3 +20,60 @@ describe("duplicate indices", () => {
     expect(errors).toEqual([]);
   });
 });
+
+describe("broken links", () => {
+  it("should not contain broken links", () => {
+    let errors = [];
+    let files = glob.sync("src/*.json");
+
+    let resources = {};
+
+    const walk_recursive = (resources, file, object, errors) => {
+      for (const property in object) {
+        if (property === "url") {
+          if (resources[object.url] === undefined) {
+            errors.push(`${file}: URL '${object.url}' not found.`);
+          } else {
+            if (resources[object.url].name !== undefined && resources[object.url].name !== object.name) {
+              errors.push(`${file}: Name mismatch for reference to '${object.url}', '${object.name}' should be '${resources[object.url].name}'`);
+            }
+
+            if (object.index !== undefined && resources[object.url].index !== object.index) {
+              errors.push(`${file}: Index mismatch for reference to '${object.url}', '${object.index}' should be '${resources[object.url].index}'`);
+            }
+          }
+        } else if (typeof object[property] === "object") {
+          walk_recursive(resources, file, object[property], errors);
+        }
+      }
+    };
+
+    for (const file of files) {
+      const fileText = fs.readFileSync(file, "utf8");
+      const fileJSON = JSON.parse(fileText);
+      fileJSON.forEach((entry) => {
```

Comment:
You might be able to extract this into a helper method? Something like this:
```js
const forEachFile(callback) {
  for (const file of files) {
      const fileText = fs.readFileSync(file, "utf8");
      const fileJSON = JSON.parse(fileText);
      fileJSON.forEach(callback);
  }
}
```

This would allow you to just say:
```js
forEachFile((entry) => {
  if (entry.url !== undefined) {
    if (entry.index === undefined) {
      errors.push(`${file}: Entity with URL '${entry.url}' should have an index.`);
    }

    resources[entry.url] = { index: entry.index, name: entry.name };
  }
});
```

and:
```js
forEachFile((entry) => {
  for (const property in entry) {
    if (typeof entry[property] === "object") {
      walk_recursive(resources, file, entry[property], errors);
    }
  }
});
```